### ISSUE TITLE
Enum mapping in provided bindings

### ISSUE THREAD

**Issue by Paul2708** [mela-org/commands#6]
What are your thoughts on adding a default mapping for enums to `ProvidedBindings`? 

**Describe the solution you'd like**
I already tried to do so [here](https://github.com/Paul2708/simple-commands/blob/d1c8829d3342eaa650f48ffd5d5aa4b3fa880b93/arguments/src/main/java/de/paul2708/commands/arguments/impl/EnumArgument.java#L39) in an other project.
In this approach the argument will be mapped by the enum constant.

**Describe alternatives you've considered**
Another idea is to assume that the enum has two methods `enum fromText(arg : String)` and `String toText()` that will be called by reflections.
I'm not sure on how the performance will be influenced by reflection calls. So another idea would be to assume that the enum implements an interface with those methods. This would only work for enums, that are written by the end-user and not existing ones (obviously).

**Comment by JohnnyJayJay** [comment]
Good idea, this should be very straight-forward to implement. I don't think there is need for reflection or additional interfaces, I'll just use `Enum.valueOf(Class, String)` and `Enum#name()` respectively. The question is whether it should always call `toUpperCase()` on the argument or not.

**Comment by Paul2708** [comment]
Well, by convention every enum constant is unique by its constant name in upper case.
Although the following code will work as well:
```java
public enum Sample {
  test,
  TEST
}
```
So calling `toUpperCase()` will match two constants..

**Comment by JohnnyJayJay** [comment]
Yes, that is the issue with this idea. On the other hand, making it very strict might result in the need to make own mapper implementations that are more convenient for end users in many cases.
I'm not really sure. I tend to the simple and strict solution. Theoretically speaking, you could also make a mapping interceptor that converts the next String to upper case or whatever is required by the enum, if you need this behaviour.

**Comment by JohnnyJayJay** [comment]
EnumMappers are now part of the provided bindings and they use strict parsing. However, I also made it easier to request the mapping of different arguments, simplifying such things as string transformers, should you need them.
The EnumMapper itself is incredibly simple, fortunately.
